src/workerd/api/crypto/crypto.c++

```diff
@@ -806,13 +806,17 @@ DigestStream::DigestStream(kj::Own<WritableStreamController> controller,
       state(Ready(kj::mv(algorithm), kj::mv(resolver))) {}
 
 void DigestStream::dispose(jsg::Lock& js) {
-  js.tryCatch([&] {
+  JSG_TRY {
     KJ_IF_SOME(ready, state.tryGet<Ready>()) {
       auto reason = js.typeError("The DigestStream was disposed.");
       ready.resolver.reject(js, reason);
       state.init<StreamStates::Errored>(js.v8Ref<v8::Value>(reason));
     }
-  }, [&](jsg::Value exception) { js.throwException(kj::mv(exception)); });
+  }
+  catch (...) {
+    jsg::Value exception = getCaughtExceptionAsJsg();
```

Comment:
This part is likely to be a common pattern... perhaps even something like...

```
JSG_CATCH(exception) {
  js.throwException(kj::mv(exception));
}
```

To boilerplate this particular piece might be handy?